### PR TITLE
cdc,pkg(ticdc): enable more linters

### LINKED ISSUE #5105: Enable some linters that find bugs
As title

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #5105

### What is changed and how it works?

Add more linters.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by zhaoxinyu** [comment]
leak test TestBatch failed. @hi-rustin PTAL

File: cdc/sink/mq_flush_worker_test.go [hunk 1]
```diff
@@ -176,6 +176,8 @@ func TestBatch(t *testing.T) {
 	batch := make([]mqEvent, 3)
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (hi-rustin) [inline cdc/sink/mq_flush_worker_test.go:179]: We should not turn on parallel here because batch was reused and the worker itself is not thread-safe.

**Review by hi-rustin — CHANGES_REQUESTED** [review]

File: cdc/server.go [hunk 2]
```diff
@@ -258,17 +258,11 @@ func (s *Server) etcdHealthChecker(ctx context.Context) error {
...
-					log.Warn("etcd health check failed", zap.Error(err))
-					cancel()
-					continue
-				}
-				_, err = httpCli.Do(req)
+				resp, err := httpCli.Get(ctx, fmt.Sprintf("%s/pd/api/v1/health", pdEndpoint))
 				if err != nil {
 					log.Warn("etcd health check error", zap.Error(err))
 				} else {
+					_ = resp.Body.Close()
```
Comment (zhaoxinyu) [inline cdc/server.go:265]:
```suggestion
		        	_, _ = io.Copy(ioutil.Discard, resp.Body)
					_ = resp.Body.Close()
```

Comment (zhaoxinyu) [inline cdc/server.go:265]: It's better to discard the response body to reuse the underlying TCP connection.

File: cdc/sink/codec/schema_registry.go [hunk 3]
```diff
@@ -337,6 +333,7 @@ func (m *AvroSchemaManager) ClearRegistry(ctx context.Context, tableName model.T
 	if err != nil {
 		return err
 	}
+	defer resp.Body.Close()
```
Comment (zhaoxinyu) [inline cdc/sink/codec/schema_registry.go:336]: Ditto

File: cdc/server.go (hunk 2 above)
Comment (overvenus) [reply cdc/server.go:265]: Is there any reference?

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/server.go (hunk 2 above)
Comment (zhaoxinyu) [reply cdc/server.go:266]: https://github.com/golang/go/blob/4d8db00641cc9ff4f44de7df9b8c4f4a4f9416ee/src/net/http/response.go#L64

File: cdc/sink/mq/mq_flush_worker_test.go [hunk 4]
```diff
@@ -173,11 +172,15 @@ func TestBatch(t *testing.T) {
...
-	var wg sync.WaitGroup
 	ctx := context.Background()
-	batch := make([]mqEvent, 3)
-	for _, test := range tests {
+	for i := range tests {
+		test := tests[i]
 		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			batch := make([]mqEvent, 3)
```
Comment (hi-rustin) [inline cdc/sink/mq/mq_flush_worker_test.go:181]: We shouldn't create a new batch here, it turns out to test its correctness when using the same block of memory.

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

**Review by hi-rustin — APPROVED** [review]
Approve in advance, only the small problem above

**Comment by hi-rustin** [comment]
/hold

Feel free unhold when https://github.com/pingcap/tiflow/pull/5106#discussion_r857451820 addressed.

**Review by hi-rustin — APPROVED** [review]
/unhold

**Pull request merged**